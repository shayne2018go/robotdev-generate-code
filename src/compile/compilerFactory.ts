import { ICodeSchema } from '@/types/ICodeSchema';
import { compileFramework } from './framework';
import { compileSyntax } from './syntax';
import parseCompileConfig from './config/parseCompileConfig';

function compilerFactory(codeSchema: ICodeSchema) {
  const compileConfig = parseCompileConfig(codeSchema);

  // 编译框架
  const frameworkTokens = compileFramework(codeSchema, compileConfig);
  // 编译语法文件
  const syntaxTokens = compileSyntax(codeSchema, compileConfig);

  const compiler = () => {
    return frameworkTokens.concat(syntaxTokens);
  };

  return {
    compiler,
  };
}

export default compilerFactory;
