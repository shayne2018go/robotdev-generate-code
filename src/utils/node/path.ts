import { resolve, dirname, relative } from 'path';

export { resolve, dirname, relative };
