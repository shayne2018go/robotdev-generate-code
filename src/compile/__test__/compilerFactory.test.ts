import compilerFactory from '@/compile/compilerFactory';
import { resolve } from 'path';
import { describe, expect, it } from 'vitest';
import codeSchema from '@/__test__/__fixture__/CodeSchema';

describe('compilerFactory', () => {
  it('compile', () => {
    const { compiler } = compilerFactory(codeSchema);

    const tokens = compiler();

    expect(tokens).toMatchSnapshot();
  });
});
