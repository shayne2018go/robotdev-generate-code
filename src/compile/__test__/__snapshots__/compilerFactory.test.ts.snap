// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compilerFactory > compile 1`] = `
[
  {
    "path": "README.md",
    "token": "# Vue 3 + TypeScript + Vite

This template should help get you started developing with Vue 3 and TypeScript in Vite. The template uses Vue 3 \`<script setup>\` SFCs, check out the [script setup docs](https://v3.vuejs.org/api/sfc-script-setup.html#sfc-script-setup) to learn more.

## Recommended IDE Setup

- [VS Code](https://code.visualstudio.com/) + [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) (and disable Vetur) + [TypeScript Vue Plugin (Volar)](https://marketplace.visualstudio.com/items?itemName=Vue.vscode-typescript-vue-plugin).

## Type Support For \`.vue\` Imports in TS

TypeScript cannot handle type information for \`.vue\` imports by default, so we replace the \`tsc\` CLI with \`vue-tsc\` for type checking. In editors, we need [TypeScript Vue Plugin (Volar)](https://marketplace.visualstudio.com/items?itemName=Vue.vscode-typescript-vue-plugin) to make the TypeScript language service aware of \`.vue\` types.

If the standalone TypeScript plugin doesn't feel fast enough to you, Volar has also implemented a [Take Over Mode](https://github.com/johnsoncodehk/volar/discussions/471#discussioncomment-1361669) that is more performant. You can enable it by the following steps:

1. Disable the built-in TypeScript Extension
   1. Run \`Extensions: Show Built-in Extensions\` from VSCode's command palette
   2. Find \`TypeScript and JavaScript Language Features\`, right click and select \`Disable (Workspace)\`
2. Reload the VSCode window by running \`Developer: Reload Window\` from the command palette.
",
  },
  {
    "path": ".gitignore",
    "token": "# Logs
logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
pnpm-debug.log*
lerna-debug.log*

node_modules
dist
dist-ssr
*.local

# Editor directories and files
.vscode/*
!.vscode/extensions.json
.idea
.DS_Store
*.suo
*.ntvs*
*.njsproj
*.sln
*.sw?
",
  },
  {
    "path": "index.html",
    "token": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <link rel=\\"icon\\" type=\\"image/svg+xml\\" href=\\"/vite.svg\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite + Vue + TS</title>
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script type=\\"module\\" src=\\"/src/main.ts\\"></script>
  </body>
</html>
",
  },
  {
    "path": "package.json",
    "token": "{
  \\"name\\": \\"vite-vue-typescript-starter\\",
  \\"private\\": true,
  \\"version\\": \\"0.0.0\\",
  \\"type\\": \\"module\\",
  \\"scripts\\": {
    \\"dev\\": \\"vite\\",
    \\"build\\": \\"vue-tsc && vite build\\",
    \\"preview\\": \\"vite preview\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^3.2.47\\"
  },
  \\"devDependencies\\": {
    \\"@vitejs/plugin-vue\\": \\"^4.1.0\\",
    \\"typescript\\": \\"^5.0.2\\",
    \\"vite\\": \\"^4.3.0-beta.2\\",
    \\"vue-tsc\\": \\"^1.2.0\\",
    \\"less\\": \\"^4.1.2\\"
  }
}
",
  },
  {
    "path": "tsconfig.json",
    "token": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"ES2020\\",
    \\"useDefineForClassFields\\": true,
    \\"module\\": \\"ESNext\\",
    \\"lib\\": [\\"ES2020\\", \\"DOM\\", \\"DOM.Iterable\\"],
    \\"skipLibCheck\\": true,

    /* Bundler mode */
    \\"moduleResolution\\": \\"bundler\\",
    \\"allowImportingTsExtensions\\": true,
    \\"resolveJsonModule\\": true,
    \\"isolatedModules\\": true,
    \\"noEmit\\": true,
    \\"jsx\\": \\"preserve\\",

    /* Linting */
    \\"strict\\": true,
    \\"noUnusedLocals\\": true,
    \\"noUnusedParameters\\": true,
    \\"noFallthroughCasesInSwitch\\": true
  },
  \\"include\\": [\\"src/**/*.ts\\", \\"src/**/*.d.ts\\", \\"src/**/*.tsx\\", \\"src/**/*.vue\\"],
  \\"references\\": [{ \\"path\\": \\"./tsconfig.node.json\\" }]
}
",
  },
  {
    "path": "tsconfig.node.json",
    "token": "{
  \\"compilerOptions\\": {
    \\"composite\\": true,
    \\"skipLibCheck\\": true,
    \\"module\\": \\"ESNext\\",
    \\"moduleResolution\\": \\"bundler\\",
    \\"allowSyntheticDefaultImports\\": true
  },
  \\"include\\": [\\"vite.config.ts\\"]
}
",
  },
  {
    "path": "vite.config.ts",
    "token": "import { defineConfig } from 'vite';
import vue from '@vitejs/plugin-vue';

export default defineConfig({
  plugins: [vue()],
  css: {
    preprocessorOptions: {
      less: {
        javascriptEnabled: true,
      },
    },
  },
});
",
  },
  {
    "path": "public/vite.svg",
    "token": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"iconify iconify--logos\\" width=\\"31.88\\" height=\\"32\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 256 257\\"><defs><linearGradient id=\\"IconifyId1813088fe1fbc01fb466\\" x1=\\"-.828%\\" x2=\\"57.636%\\" y1=\\"7.652%\\" y2=\\"78.411%\\"><stop offset=\\"0%\\" stop-color=\\"#41D1FF\\"></stop><stop offset=\\"100%\\" stop-color=\\"#BD34FE\\"></stop></linearGradient><linearGradient id=\\"IconifyId1813088fe1fbc01fb467\\" x1=\\"43.376%\\" x2=\\"50.316%\\" y1=\\"2.242%\\" y2=\\"89.03%\\"><stop offset=\\"0%\\" stop-color=\\"#FFEA83\\"></stop><stop offset=\\"8.333%\\" stop-color=\\"#FFDD35\\"></stop><stop offset=\\"100%\\" stop-color=\\"#FFA800\\"></stop></linearGradient></defs><path fill=\\"url(#IconifyId1813088fe1fbc01fb466)\\" d=\\"M255.153 37.938L134.897 252.976c-2.483 4.44-8.862 4.466-11.382.048L.875 37.958c-2.746-4.814 1.371-10.646 6.827-9.67l120.385 21.517a6.537 6.537 0 0 0 2.322-.004l117.867-21.483c5.438-.991 9.574 4.796 6.877 9.62Z\\"></path><path fill=\\"url(#IconifyId1813088fe1fbc01fb467)\\" d=\\"M185.432.063L96.44 17.501a3.268 3.268 0 0 0-2.634 3.014l-5.474 92.456a3.268 3.268 0 0 0 3.997 3.378l24.777-5.718c2.318-.535 4.413 1.507 3.936 3.838l-7.361 36.047c-.495 2.426 1.782 4.5 4.151 3.78l15.304-4.649c2.372-.72 4.652 1.36 4.15 3.788l-11.698 56.621c-.732 3.542 3.979 5.473 5.943 2.437l1.313-2.028l72.516-144.72c1.215-2.423-.88-5.186-3.54-4.672l-25.505 4.922c-2.396.462-4.435-1.77-3.759-4.114l16.646-57.705c.677-2.35-1.37-4.583-3.769-4.113Z\\"></path></svg>",
  },
  {
    "path": "src/App.vue",
    "token": "<script setup lang=\\"ts\\">
import HelloWorld from './components/HelloWorld.vue';
</script>

<template>
  <div>
    <a href=\\"https://vitejs.dev\\" target=\\"_blank\\">
      <img src=\\"/vite.svg\\" class=\\"logo\\" alt=\\"Vite logo\\" />
    </a>
    <a href=\\"https://vuejs.org/\\" target=\\"_blank\\">
      <img src=\\"./assets/vue.svg\\" class=\\"logo vue\\" alt=\\"Vue logo\\" />
    </a>
  </div>
  <HelloWorld msg=\\"Vite + Vue\\" />
</template>

<style scoped>
.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.vue:hover {
  filter: drop-shadow(0 0 2em #42b883aa);
}
</style>
",
  },
  {
    "path": "src/main.ts",
    "token": "import { createApp } from 'vue'
import './style.css'
import App from './App.vue'

createApp(App).mount('#app')
",
  },
  {
    "path": "src/style.css",
    "token": ":root {
  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
  line-height: 1.5;
  font-weight: 400;

  color-scheme: light dark;
  color: rgba(255, 255, 255, 0.87);
  background-color: #242424;

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
}

a {
  font-weight: 500;
  color: #646cff;
  text-decoration: inherit;
}
a:hover {
  color: #535bf2;
}

body {
  margin: 0;
  display: flex;
  place-items: center;
  min-width: 320px;
  min-height: 100vh;
}

h1 {
  font-size: 3.2em;
  line-height: 1.1;
}

button {
  border-radius: 8px;
  border: 1px solid transparent;
  padding: 0.6em 1.2em;
  font-size: 1em;
  font-weight: 500;
  font-family: inherit;
  background-color: #1a1a1a;
  cursor: pointer;
  transition: border-color 0.25s;
}
button:hover {
  border-color: #646cff;
}
button:focus,
button:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}

.card {
  padding: 2em;
}

#app {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

@media (prefers-color-scheme: light) {
  :root {
    color: #213547;
    background-color: #ffffff;
  }
  a:hover {
    color: #747bff;
  }
  button {
    background-color: #f9f9f9;
  }
}
",
  },
  {
    "path": "src/vite-env.d.ts",
    "token": "/// <reference types=\\"vite/client\\" />
",
  },
  {
    "path": "src/assets/vue.svg",
    "token": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"iconify iconify--logos\\" width=\\"37.07\\" height=\\"36\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 256 198\\"><path fill=\\"#41B883\\" d=\\"M204.8 0H256L128 220.8L0 0h97.92L128 51.2L157.44 0h47.36Z\\"></path><path fill=\\"#41B883\\" d=\\"m0 0l128 220.8L256 0h-51.2L128 132.48L50.56 0H0Z\\"></path><path fill=\\"#35495E\\" d=\\"M50.56 0L128 133.12L204.8 0h-47.36L128 51.2L97.92 0H50.56Z\\"></path></svg>",
  },
  {
    "path": "src/components/HelloWorld.vue",
    "token": "<script setup lang=\\"ts\\">
import { ref } from 'vue'

defineProps<{ msg: string }>()

const count = ref(0)
</script>

<template>
  <h1>{{ msg }}</h1>

  <div class=\\"card\\">
    <button type=\\"button\\" @click=\\"count++\\">count is {{ count }}</button>
    <p>
      Edit
      <code>components/HelloWorld.vue</code> to test HMR
    </p>
  </div>

  <p>
    Check out
    <a href=\\"https://vuejs.org/guide/quick-start.html#local\\" target=\\"_blank\\"
      >create-vue</a
    >, the official Vue + Vite starter
  </p>
  <p>
    Install
    <a href=\\"https://github.com/vuejs/language-tools\\" target=\\"_blank\\">Volar</a>
    in your IDE for a better DX
  </p>
  <p class=\\"read-the-docs\\">Click on the Vite and Vue logos to learn more</p>
</template>

<style scoped>
.read-the-docs {
  color: #888;
}
</style>
",
  },
]
`;
